src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicDocumentPullDiagnosticsHandler.cs

```diff
@@ -97,7 +97,11 @@ protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagno
         var nonLocalDocumentDiagnostics = diagnosticParams.Identifier == DocumentNonLocalDiagnosticIdentifier.ToString();
 
         // Task list items are not reported through the public LSP diagnostic API.
-        return ValueTaskFactory.FromResult(DocumentPullDiagnosticHandler.GetDiagnosticSources(DiagnosticKind.All, nonLocalDocumentDiagnostics, taskList: false, context, GlobalOptions));
+        var source = nonLocalDocumentDiagnostics
+            ? DocumentPullDiagnosticHandler.GetNonLocalDiagnosticSource(context, GlobalOptions)
+            : DocumentPullDiagnosticHandler.GetDiagnosticSource(DiagnosticKind.All, context);
```

Comment:
@tmat @dibarbet  what is "DocumentNonLocalDiagnosticIdentifier"?

Reply:
oh - we already create buckets in VSCode - I completely forgot about this.  Its for diagnostics reported on the document during compilation end
https://github.com/dotnet/roslyn/blob/b67df4c5e39ac1a104b2e573542211d4d6b1cd1d/src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicDocumentPullDiagnosticsHandler_IOnInitialized.cs#L43